sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/IPartitionProcessor.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Processes events received from the Azure Event Hubs service. A single instance of a class implementing this
+    ///   interface will be created for every partition the associated <see cref="EventProcessor" /> owns.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   Every aforementioned instance is created by an <see cref="IPartitionProcessorFactory" /> provided by
+    ///   the user in the <see cref="EventProcessor" /> constructor.
+    /// </remarks>
+    ///
+    public interface IPartitionProcessor
```

Comment:
Contrary to the partition manager, I think an interface is appropriate and justifiable here.   It's unlikely to change, the purpose is to process events which could be an extension of an existing event processor - and this was a case that the service team highlighted as feedback from users.